kong/plugins/influxdb-log/handler.lua

```diff
@@ -0,0 +1,98 @@
+local HttpLogHandler = require "kong.plugins.http-log.handler"
+local cjson = require "cjson"
+
+local InfluxdbLogHandler = HttpLogHandler:extend()
+
+InfluxdbLogHandler.PRIORITY = 1
+
+-- influxdb util function
+-- ref:https://docs.influxdata.com/influxdb/v0.13/write_protocols/write_syntax/
+local function influxdb_point(ngx)
+  return {
+    tag = {
+      uri = ngx.var.request_uri,
+      request_uri = ngx.var.scheme.."://"..ngx.var.host..":"..ngx.var.server_port..ngx.var.request_uri,
+      request_method = ngx.req.get_method(), -- http method
+      response_status = ngx.status,
+      client_ip = ngx.var.remote_addr,
+      api_id = ngx.ctx.api.id,
+      authenticated_entity_id = ngx.ctx.authenticated_credential and ngx.ctx.authenticated_credential.id,
+      authenticated_entity_consumer_id = ngx.ctx.authenticated_credential and ngx.ctx.authenticated_credential.consumer_id
+    },
+    field = {
+      request_size = ngx.var.request_length,
+      response_size = ngx.var.bytes_sent,
+      latencies_kong = (ngx.ctx.KONG_ACCESS_TIME or 0) +
+               (ngx.ctx.KONG_RECEIVE_TIME or 0),
+      latencies_proxy = ngx.ctx.KONG_WAITING_TIME or -1,
+      latencies_request = ngx.var.request_time * 1000,
+      started_at = ngx.req.start_time() * 1000
+    }
+  }
+end
+
+local function influxdb_escape(str)
+    if type(str) ~= "string" then
+        return str
+    end
+
+    return string.gsub(str, "[=, ]", function (w) return "\\"..w end)
+
+end
+
+-- Generates influxdb line
+-- @param `measurement`, similar to MySQL table name
+-- @param `message`  Message to be logged
+-- @return `line` of influxdb line protocol
+local function generate_influxdb_line(measurement, message)
+    local line = measurement
```

Comment:
besides caching functions etc. I think this would be more performant if it generated a table/list and only performed a single `concat` operation at the end